File: src/php/core/src/Style/MetaCreator.php
<?php

declare(strict_types=1);

namespace DiceBear\Style;

/**
 * Read-only view over the `meta.creator` block of a style definition.
 */
class MetaCreator
{
    /**
     * @param array<string, mixed> $data
     */
    public function __construct(private readonly array $data) {}

    /**
     * Returns the creator's display name, or `null` when not set.
     */
    public function name(): ?string
    {
        return $this->data['name'] ?? null;
    }

    /**
     * Returns the creator's homepage URL, or `null` when not set.
     */
    public function url(): ?string
    {
        return $this->data['url'] ?? null;
    }
}

File: src/rust/core/src/style/meta_creator.rs
//! Read-only view over a style definition's `meta.creator` block.

use serde::Deserialize;

#[derive(Deserialize, Default)]
pub(crate) struct MetaCreator {
    name: Option<String>,
    url: Option<String>,
}

impl MetaCreator {
    pub(crate) fn name(&self) -> Option<&str> {
        self.name.as_deref()
    }

    #[allow(dead_code)]
    pub(crate) fn url(&self) -> Option<&str> {
        self.url.as_deref()
    }
}
